setup.py

```diff
@@ -64,7 +64,7 @@
 
     # Allows `setup.py test` to work correctly with pytest
     setup_requires=["numpy>=1.21.2",
-                    "scipy>=1.7.3",
+                    "scipy==1.11.1",
```

Comment:
The most updated version of `scipy` leads to a testing error because `scipy` started to usa `hypothesis` for property-based tests, 63d3d845dd1df071b1fde0882002e202a7e2e849. The full link
is https://github.com/scipy/scipy/tree/63d3d845dd1df071b1fde0882002e202a7e2e849. This leads to the error information below. Fixing the `scipy` addresses the problem. Wecan wait until `scipy` fixes this problem.
@FarnazH 

```
________________________ ERROR collecting test session _________________________
  .tox/qa/lib/python3.11/site-packages/_pytest/config/__init__.py:642: in _importconftest
      mod = import_path(conftestpath, mode=importmode, root=rootpath)
          conftestpath = PosixPath('/home/runner/work/DiverseSelector/DiverseSelector/.eggs/scipy-1.11.3-py3.11-linux-x86_64.egg/scipy/conftest.py')
          exc_info   = (<class '_pytest.pathlib.ImportPathMismatchError'>, ImportPathMismatchError('scipy.conftest', '/home/runner/work/Diver.../DiverseSelector/.eggs/scipy-1.11.3-py3.11-linux-x86_64.egg/scipy/conftest.py')), <traceback object at 0x7f650156f600>)
          existing   = None
          importmode = 'prepend'
          pkgpath    = PosixPath('/home/runner/work/DiverseSelector/DiverseSelector/.eggs/scipy-1.11.3-py3.11-linux-x86_64.egg/scipy')
          rootpath   = PosixPath('/home/runner/work/DiverseSelector/DiverseSelector')
          self       = <_pytest.config.PytestPluginManager object at 0x7f650a1f29d0>
  .tox/qa/lib/python3.11/site-packages/_pytest/pathlib.py:588: in import_path
      raise ImportPathMismatchError(module_name, module_file, path)
  E   _pytest.pathlib.ImportPathMismatchError: ('scipy.conftest', '/home/runner/work/DiverseSelector/DiverseSelector/.tox/qa/lib/python3.11/site-packages/scipy/conftest.py', PosixPath('/home/runner/work/DiverseSelector/DiverseSelector/.eggs/scipy-1.11.3-py3.11-linux-x86_64.egg/scipy/conftest.py'))
          ignore     = ''
          is_same    = False
          mod        = <module 'scipy.conftest' from '/home/runner/work/DiverseSelector/DiverseSelector/.tox/qa/lib/python3.11/site-packages/scipy/conftest.py'>
          mode       = <ImportMode.prepend: 'prepend'>
          module_file = '/home/runner/work/DiverseSelector/DiverseSelector/.tox/qa/lib/python3.11/site-packages/scipy/conftest.py'
          module_name = 'scipy.conftest'
          names      = ['scipy', 'conftest']
          p          = PosixPath('/home/runner/work/DiverseSelector/DiverseSelector/.eggs/scipy-1.11.3-py3.11-linux-x86_64.egg/scipy/conftest.py')
          path       = PosixPath('/home/runner/work/DiverseSelector/DiverseSelector/.eggs/scipy-1.11.3-py3.11-linux-x86_64.egg/scipy/conftest.py')
          pkg_path   = PosixPath('/home/runner/work/DiverseSelector/DiverseSelector/.eggs/scipy-1.11.3-py3.11-linux-x86_64.egg/scipy')
          pkg_root   = PosixPath('/home/runner/work/DiverseSelector/DiverseSelector/.eggs/scipy-1.11.3-py3.11-linux-x86_64.egg')
          root       = PosixPath('/home/runner/work/DiverseSelector/DiverseSelector')
```